drivers/adc/adc_stm32.c

```diff
@@ -1174,6 +1176,34 @@ static const struct adc_driver_api api_stm32_driver_api = {
 	.ref_internal = STM32_ADC_VREF_MV, /* VREF is usually connected to VDD */
 };
 
+/* st_prescaler property requires 2 elements : clock ASYNC/SYNC and DIV */
+#define STM32_ADC_CLOCK(x, prop) DT_INST_PROP_BY_IDX(x, prop, 0)
+#define STM32_ADC_DIV(x, prop) DT_INST_PROP_BY_IDX(x, prop, 1)
+
+/* macro to check is 1st element is SYNC or ASYNC */
+#define STM32_ADC_CLOCK_IS_SYNC(x, prop) IS_EQ(STM32_ADC_CLOCK(x, prop), SYNC)
+
+/* Macro to set the prefix depending on the 1st element */
+#define STM32_ADC_DO_PREFIX(x, prop)				\
+	COND_CODE_1(IS_EQ(STM32_ADC_CLOCK(x, prop), SYNC),	\
+		prescaler_sync,					\
+		prescaler_async)
+
+/* Concat prefix (1st element) and DIV value (2nd element) of st,prescaler property */
+#if defined(CONFIG_SOC_SERIES_STM32F1X)
+/* not available for that serie */
+#define DT_INST_PRESC_PROP(x, prop) 0
+#else
+/*
+ * The st,prescaler property must declare 2 elements
+ * else choosing the default prescaler clock and div by 4
+ */
+#define DT_INST_PRESC_PROP(x, prop)					\
+	COND_CODE_1(IS_EQ(DT_INST_PROP_LEN(x, prop), 2),		\
```

Comment:
What about ?
```suggestion
	COND_CODE_1(DT_INST_NODE_HAS_PROP(x, st_adc_prescaler),		\
```

Reply:
COND_CODE_1(CONFIG_FLAG, ...) accepts CONFIG_FLAG to be IS_EQ(something) and  IS_EQ(property len, 2) is a valid CONFIG_FLAG
--> DT_INST_NODE_HAS_PROP(x, st_adc_prescaler) is not a _CONFIG_FLAG_ for COND_CODE_1

Reply:
next step: add a check on st,adc-prescaler len and 1st element to be SYNC or ASYNC if present

Reply:
Weird, there are >50 occurences of `COND_CODE_1(DT_INST_NODE_HAS_PROP(` in the tree.

Reply:
ok with COND_CODE_1(DT_INST_NODE_HAS_PROP()